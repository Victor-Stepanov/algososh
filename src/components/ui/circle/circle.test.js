import React from "react";
import TestRenderer from 'react-test-renderer';
import {Circle} from "./circle";
import {ElementStates} from "../../../types/element-states";

describe('Тестирование компонента Circle', () => {
    it('проверка компонента без буквы', () => {
        const circle = TestRenderer
            .create(<Circle/>)
            .toJSON()
        expect(circle).toMatchSnapshot()
    })
    it('проверка компонента с буквами', () => {
        const circle = TestRenderer
            .create(<Circle letter={'hello'}/>)
            .toJSON()
        expect(circle).toMatchSnapshot()
    })
    it('проверка компонента с head', () => {
        const circle = TestRenderer
            .create(<Circle head={'1'}/>)
            .toJSON()
        expect(circle).toMatchSnapshot()
    })
    it('проверка компонента с react-элементом в head', () => {
        const circle = TestRenderer
            .create(<Circle head={<Circle/>}/>)
            .toJSON()
        expect(circle).toMatchSnapshot()
    })
    it('проверка компонента с tail', () => {
        const circle = TestRenderer
            .create(<Circle tail={'-1'}/>)
            .toJSON()
        expect(circle).toMatchSnapshot()
    })
    it('проверка компонента c react-элементом в tail', () => {
        const circle = TestRenderer
            .create(<Circle tail={<Circle/>}/>)
            .toJSON()
        expect(circle).toMatchSnapshot()
    })
    it('проверка компонента с index', () => {
        const circle = TestRenderer
            .create(<Circle index={0}/>)
            .toJSON()
        expect(circle).toMatchSnapshot()
    })
    it('проверка компонента с пропом isSmall ===  true', () => {
        const circle = TestRenderer
            .create(<Circle isSmall={true}/>)
            .toJSON()
        expect(circle).toMatchSnapshot()
    })
    it('проверка компонента в состоянии default', () => {
        const circle = TestRenderer
            .create(<Circle state={ElementStates.Default}/>)
            .toJSON()
        expect(circle).toMatchSnapshot()
    })
    it('проверка компонента в состоянии changing', () => {
        const circle = TestRenderer
            .create(<Circle state={ElementStates.Changing}/>)
            .toJSON()
        expect(circle).toMatchSnapshot()
    })
    it('проверка компонента в состоянии modified', () => {
        const circle = TestRenderer
            .create(<Circle state={ElementStates.Modified}/>)
            .toJSON()
        expect(circle).toMatchSnapshot()
    })
})