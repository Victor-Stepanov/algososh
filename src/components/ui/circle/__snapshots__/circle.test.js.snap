// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование компонента Circle проверка компонента c react-элементом в tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Тестирование компонента Circle проверка компонента без буквы 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle проверка компонента в состоянии changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle проверка компонента в состоянии default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle проверка компонента в состоянии modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle проверка компонента с head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    1
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle проверка компонента с index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    0
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Тестирование компонента Circle проверка компонента с react-элементом в head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle проверка компонента с tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    -1
  </div>
</div>
`;

exports[`Тестирование компонента Circle проверка компонента с буквами 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      hello
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle проверка компонента с пропом isSmall ===  true 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
